lib/linguist/strategy/extension.rb

```diff
@@ -1,5 +1,10 @@
 module Linguist
   module Strategy
+    # Extensions considered too common for simple extension lookup.
+    # Languages which use such an extension must instead satisfy a
+    # heuristic (or some other strategy) to be matched on extension.
+    GENERIC = %w[.sol]
```

Comment:
This list should be in `languages.yml` instead.

Reply:
Why? I fail to see how this property is semantically different from being `vendored`, save the fact we have enough [vendored patterns](https://github.com/github/linguist/blob/master/lib/linguist/vendor.yml) to warrant its own file. Since we only have one extension marked as generic, having its own file would be overkill.

Reply:
Given this effectively implements a not-a-language option for the heuristic strategy only, I doubt we'll have to wait long before we see new languages added here. I'm not suggesting a new file, but simply a new entry in `languages.yml`. I don't want this list to be buried where we will forget it. It's also easier to actually use the list (e.g., for additional tests) if it's in `languages.yml`.

Reply:
> but simply a new entry in `languages.yml`

And where do you propose I put it? The top-level entry of the file is an array of language objects…

Reply:
I think it should be attached to specific languages and we should have a test to check the heuristic rules only pick among those languages.

Reply:
Can you provide a code sample of what you're talking about? I'm still confused.

Reply:
Sure. Something like:
```yaml
Solidity:
  type: programming
  color: "#AA6746"
  extensions:
  - .sol
  generic_extensions:
  - .sol
  ace_mode: text
  tm_scope: source.solidity
  language_id: 237469032
```

Reply:
Not only is this repeating information we already have, it's visibly confusing to users as to which extensions-list requires updating. Moreover, we need to do the same for *every* language that uses a generic extension — it's more work for users, has an increased risk of complications, and at the end of the day, this approach has **zero benefits** over simply hardcoding the list somewhere else.

Reply:
I think we're OK keeping this list here for now. We can explore pulling it out to its own file if we see the demand for it. Excluding the numbered files @Alhadis has mentioned, I know `.k` will likely be a candidate for this list.

Reply:
Sure, let's keep here until we need it in a separate file. I doubt it will be long before we need to refer to this list in specific tests though...